task/delete.go

```diff
@@ -0,0 +1,93 @@
+package task
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-api-client"
+	"github.com/livepeer/livepeer-data/pkg/data"
+	"golang.org/x/sync/errgroup"
+)
+
+func TaskDelete(tctx *TaskContext) (*TaskHandlerOutput, error) {
+
+	var (
+		ctx    = tctx.Context
+		asset  = tctx.InputAsset
+		osSess = tctx.outputOS
+	)
+
+	directory := asset.PlaybackID
+
+	files, err := osSess.ListFiles(ctx, directory, "/")
+
+	if err != nil {
+		glog.Errorf("Error listing files in directory %v", directory)
+	}
+
+	totalFiles := files.Files()
+
+	accumulator := NewAccumulator()
+	tctx.Progress.TrackCount(accumulator.Size, uint64(len(totalFiles)), 0)
+
+	const maxRetries = 3
+	const retryInterval = time.Second
+
+	eg := errgroup.Group{}
+
+	for _, file := range totalFiles {
+		filename := file.Name
+		eg.Go(func() error {
+			var err error
+			for i := 0; i < maxRetries; i++ {
+				err = osSess.DeleteFile(ctx, filename)
+				if err == nil {
+					accumulator.Accumulate(1)
+					return nil
+				}
+				glog.Errorf("Error deleting file %v: %v (retrying...)", filename, err)
+				time.Sleep(retryInterval)
+			}
+			return fmt.Errorf("failed to delete file %v after %d retries", filename, maxRetries)
```

Comment:
If you return an error the whole error group will cancel. You should always return `nil` and only log the errors instead (if you want to delete as much as possible before failing)